src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/validators/PathValidator.java

```diff
@@ -0,0 +1,456 @@
+ */
+
+package org.openkilda.wfm.topology.nbworker.validators;
+
+import static org.openkilda.model.PathComputationStrategy.COST_AND_AVAILABLE_BANDWIDTH;
+import static org.openkilda.model.PathComputationStrategy.LATENCY;
+import static org.openkilda.model.PathComputationStrategy.MAX_LATENCY;
+
+import org.openkilda.messaging.info.network.PathValidationResult;
+import org.openkilda.model.Flow;
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.Isl;
+import org.openkilda.model.IslStatus;
+import org.openkilda.model.PathSegment;
+import org.openkilda.model.PathValidationData;
+import org.openkilda.model.SwitchProperties;
+import org.openkilda.persistence.repositories.FlowRepository;
+import org.openkilda.persistence.repositories.IslRepository;
+import org.openkilda.persistence.repositories.SwitchPropertiesRepository;
+import org.openkilda.persistence.repositories.SwitchRepository;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class PathValidator {
+
+    private final IslRepository islRepository;
+    private final FlowRepository flowRepository;
+    private final SwitchPropertiesRepository switchPropertiesRepository;
+
+    private final SwitchRepository switchRepository;
+
+    public PathValidator(IslRepository islRepository,
+                         FlowRepository flowRepository,
+                         SwitchPropertiesRepository switchPropertiesRepository,
+                         SwitchRepository switchRepository) {
+        this.islRepository = islRepository;
+        this.flowRepository = flowRepository;
+        this.switchPropertiesRepository = switchPropertiesRepository;
+        this.switchRepository = switchRepository;
+    }
+
+    /**
+     * Validates whether it is possible to create a path with the given parameters. When there are obstacles, this
+     * validator returns all errors found on each segment. For example, when there is a path 1-2-3-4 and there is
+     * no link between 1 and 2, no sufficient latency between 2 and 3, and not enough bandwidth between 3-4, then
+     * this validator returns all 3 errors.
+     *
+     * @param pathValidationData path parameters to validate.
+     * @return a response object containing the validation result and errors if any
+     */
+    public PathValidationResult validatePath(PathValidationData pathValidationData) {
+        Set<String> result = pathValidationData.getPathSegments().stream()
+                .map(segment -> executeValidations(
+                        new InputData(pathValidationData, segment),
+                        new RepositoryData(islRepository, flowRepository, switchPropertiesRepository, switchRepository),
```

Comment:
Actually, this class is just a holder for several references. This sort of Java objects are cheap. I can move the contents of that class to a member variables, there will be no RepositoryData at all. But I created it like this, so that all validation methods could be parts of a stream and I chose BiFunction with encapsulated repos, because I didn't know which repos I will require here. Also, I can unwrap the 4 repos into a new functional interface.